File: solutions/c/resistor-color/1/resistor_color.c
#include "resistor_color.h"

int color_code( resistor_band_t color ) {
    return color;
}

int * colors() {
    static int colors[] = {
        BLACK,
        BROWN,
        RED,
        ORANGE,
        YELLOW,
        GREEN,
        BLUE,
        VIOLET,
        GREY,
        WHITE
    };
    return colors;
}

File: solutions/typescript/resistor-color/1/resistor-color.ts
export const colorCode = ( colorName : string ) => {
    let colorCode = -1;
    for ( var i = 0; i < COLORS.length; i++ ) {
        if ( colorName == COLORS[i] ) {
            colorCode = i;
            break;
        }
    }
    return colorCode;
}

export const COLORS = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
];
